### ISSUE TITLE
Summarizing and subtotaling tokens

### ISSUE THREAD

**Issue by cspenn** [kbenoit/quanteda#1055]
What's the best way to create a summarized subtotal of tokens?

What I'm trying to do is take a text, tokenize large multi-word strings, then count up the frequencies of those strings.

```
tokentext <- tokens(doc.corpus,remove_punct=TRUE,remove_symbols=TRUE,remove_numbers=TRUE,remove_url=TRUE,remove_twitter=TRUE,verbose=TRUE)
compoundtext <- tokens_compound(tokentext, phrase(dict), concatenator="_",case_insensitive=TRUE,join=TRUE)
ngrams <- tokens_ngrams(compoundtext, n = c(5,12), skip = 0, concatenator = "_")
countgrams <- ntoken(ngrams)
```

This gets me the ngrams object with windowed 5-12 word length character strings, but ntoken just produces a large mess.

Example output:
```
text27596 | 201
-- | --
text28144 | 197
text27300 | 195
text27932 | 195
text28156 | 195
```

What the ideal output would be:

```
I_wish_I_lived_in_a_nice_house | 254
I_wish_I_lived_in_a_bigger_house| 213
```

etc.

How do I do this?

Thank you!

**Comment by koheiw** [comment]
Like this?
```r
require(quanteda)
toks <- tokens(data_corpus_inaugural, ngrams = 5:12, remove_punct = TRUE)
as.data.frame(topfeatures(dfm(toks)))
```
```
                                      topfeatures(dfm(toks))
constitution_of_the_united_states                         20
the_constitution_of_the_united                            19
the_constitution_of_the_united_states                     19
to_the_best_of_my                                         10
people_of_the_united_states                               10
of_the_united_states_to                                   10
the_best_of_my_ability                                     9
the_people_of_the_united                                   9
to_the_best_of_my_ability                                  9
the_nations_of_the_world                                   9
```

**Comment by cspenn** [comment]
That's awesome- thank you!

**Comment by kbenoit** [comment]
There’s also `textstat_frequency()`:


```r

> toks <- tokens(data_corpus_inaugural[1:2], ngrams = 5:12, remove_punct = TRUE)
> dfm(toks) %>% textstat_frequency() %>% head()
                            feature frequency rank docfreq
1           the_voice_of_my_country         2    1       2
2 fellow-citizens_of_the_senate_and         1    2       1
3        fellow_citizens_i_am_again         1    3       1
4              of_the_senate_and_of         1    4       1
5             the_senate_and_of_the         1    5       1
6           senate_and_of_the_house         1    6       1

```

On 7 Nov 2017, at 20:02, Christopher Penn <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #1055<https://github.com/kbenoit/quanteda/issues/1055>.
